pages/api/feedback.ts

```diff
@@ -0,0 +1,60 @@
+import type { NextApiRequest, NextApiResponse } from "next"
+import aws from "aws-sdk"
+
+import { appConfig } from "../../src/config/config"
+import {
+  getFeedbackEmailText,
+  getFeedbackEmailHTML,
+} from "../../src/utils/feedbackUtils"
+import { encodeHTML } from "../../src/utils/appUtils"
+
+async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method === "POST") {
+    if (!req.body) res.status(400).json({ error: "Malformed request" })
+    aws.config.update({ region: "us-east-1" })
+
+    const fields = JSON.parse(req.body)
+
+    const fullUrl = encodeHTML(req.headers.referer)
+
+    const emailText = getFeedbackEmailText(fullUrl, fields)
+    const emailHTML = getFeedbackEmailHTML(fullUrl, fields)
+
+    const emailParams = {
```

Comment:
is there a way to leverage typescript to detail the required fields here? It's not immediately clear to me what part of this object is required and what is requiring it.